packages/experiment-codec/tests/test-vector/testUtility.ts

```diff
@@ -0,0 +1,250 @@
+      type: "vector",
+      itemCodec,
+    },
+    ...codec,
+  };
+};
+export const testOption = (itemCodec: BytesCodec): TestBytesCodec => {
+  const codec = option(itemCodec);
+  return {
+    testMetadata: {
+      type: "option",
+      itemCodec,
+    },
+    ...codec,
+  };
+};
+
+export function generateDefaultCodecData(codec: any): any {
+  const testMetadata = codec.testMetadata;
+  if (testMetadata.type === "array") {
+    return new Array(testMetadata.itemCount).fill(
+      generateDefaultCodecData(testMetadata.itemCodec)
+    );
+  }
+  if (testMetadata.type === "struct" || testMetadata.type === "table") {
+    const defaultValue = {};
+    (testMetadata.itemCodecs as Array<[string, BytesCodec]>).forEach(
+      ([field, codec]) => {
+        Object.assign(defaultValue, {
+          [field]: generateDefaultCodecData(codec),
+        });
+      }
+    );
+    return defaultValue;
+  }
+  if (testMetadata.type === "option") {
+    return undefined;
+  }
+  if (testMetadata.type === "union") {
+    const itemCodec = testMetadata.itemCodecs[0];
+    return {
+      type: itemCodec[0],
+      value: generateDefaultCodecData(itemCodec[1]),
+    };
+  }
+  if (testMetadata.type === "vector") {
+    return [];
+  }
+  if (testMetadata.type === "byte") {
+    return 0;
+  }
+}
+
+export function fullfillPartialCodecData(
+  codec: any,
+  data: any,
+  item: any
+): any {
+  const defaultData = generateDefaultCodecData(codec);
+  const testMetadata = codec.testMetadata;
+  if (testMetadata.type === "array") {
+    const fulfilledData = defaultData;
+    Object.entries(data).forEach(([index, value]) => {
+      fulfilledData[(index as any) as number] = value;
+    });
+    return fulfilledData;
+  }
+  if (testMetadata.type === "struct" || testMetadata.type === "table") {
+    const fulfilledData = defaultData;
+    Object.entries(data).forEach(([key, value]) => {
+      const itemCodec = (testMetadata.itemCodecs as Array<
+        [string, BytesCodec]
+      >).find(([field]) => field === key)![1];
+      fulfilledData[key] = itemCodec.unpack(toArrayBuffer(value as any));
+    });
+    return fulfilledData;
+  }
+  if (testMetadata.type === "option") {
+    return codec.unpack(toArrayBuffer(item));
+  }
+  if (testMetadata.type === "union") {
+    const itemCodec = (testMetadata.itemCodecs as Array<
+      [string, BytesCodec]
+    >).find(([field]) => field === item.type)![1];
+    const fulfilledData = itemCodec.unpack(toArrayBuffer(item.data));
+    return {
+      type: item.type,
+      value: fulfilledData,
+    };
+  }
+  if (testMetadata.type === "vector") {
+    return (data as Array<string>).map((line) =>
+      testMetadata.itemCodec.unpack(toArrayBuffer(line))
+    );
+  }
+  if (testMetadata.type === "byte") {
+    return 0;
+  }
+}
+
+type TestCase = {
+  name: string;
+  data?: any;
+  item?: any;
+  expected: string;
+};
+
+export const loadTests = (path: string) => {
+  let cases: Array<TestCase> = [];
+  try {
```

Comment:
We should not use try-catch with a large block, which may make it difficult to troubleshoot when something goes wrong

Reply:
moved all codes out except for io operations